packages/pg/lib/connection-parameters.js

```diff
@@ -25,15 +25,31 @@ var val = function (key, config, envVar) {
   return config[key] || envVar || defaults[key]
 }
 
-var useSsl = function () {
-  switch (process.env.PGSSLMODE) {
+var normalizeSSLConfig = function (modeFromConfig) {
+  // if the ssl parameter passed to config is not a string, just return it
+  // directly (it will be passed directly to tls.connect)
+  // this way you can pass all the ssl params in via constructor:
+  // new Client({ ssl: { minDHSize: 1024 } }) etc
+  if (modeFromConfig !== undefined && typeof modeFromConfig !== 'string') {
+    return modeFromConfig
+  }
+  const mode = modeFromConfig || process.env.PGSSLMODE
+
+  switch (mode) {
     case 'disable':
       return false
     case 'prefer':
     case 'require':
     case 'verify-ca':
     case 'verify-full':
       return true
+    // no-verify is not standard to libpq but allows specifying
```

Comment:
If the intent is to follow the libpq behavior, then `prefer` should also skip certificate validation, and `require` only needs to validate the certificate if the root CA file is available.
Also, the `checkServerIdentity` should be disabled (set to no-op) if the mode is not `verify-full`.

Reply:
Yeah totally...but I'm probably not gonna change these - it's a semver major to change these and they've behaved this way since...kinda forever w/ `pg`.  Slight inconsistencies w/ the native bindings aren't ideal, but they're less not-ideal than breaking changes.  I will do a better job of documenting these options though in the docs and what they mean and how they diverge.